### DISCUSSION TITLE
GitHub Actions running old Workflow file?

### DISCUSSION THREAD

**Discussion by edemaine** [community/community#122045] [Actions] [answered]
### Select Topic Area

Question

### Body

Are there scenarios where GitHub Actions runs an older version of a workflow, instead of the current version in a PR?

Here's an example where this seems to be happening:
* [Workflow run](https://github.com/KaTeX/KaTeX/actions/runs/8927743117/job/24521789025?pr=3947)
* [Workflow file](https://github.com/KaTeX/KaTeX/actions/runs/8927743117/workflow?pr=3947) has a step "Install dependencies" that doesn't seem to be getting run
* [PR](https://github.com/KaTeX/KaTeX/pull/3947)

I've tried rerunning the jobs, and modifying the workflow file a few more times, and it seems to be making no difference. Is there a cache somewhere I should clear? I've tried deleting everything in https://github.com/KaTeX/KaTeX/actions/caches

**Comment by GC-Configit** [comment]
We have a similar issue. In the pull request, we have updated the workflow file, but the runner is using the old file. Furthermore, if you open a workflow run ➡ summary  ➡ Workflow file, you can see a changed file with new code but in the workflow run you can't spot these changes it still runs the old file. 

However, the changes were applied when I ran the same workflow using dispatch action. It seems that PR GitHub uses old workflow files, but I have no idea why.

**Reply by edemaine** [reply]
Yes, the behavior sounds identical — thank you for sharing. If I understand [dispatch actions](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow?tool=cli), it seems like I need to enable `workflow_dispatch` on the default branch, and this can't be done purely within the PR. 🙁

**Reply by GC-Configit** [reply, marked as answer]
@edemaine ok, I found my issue. In that specific workflow file, I was using `pull_request_target` trigger instead of `pull_request`. This caused this behaviour I copy pasted it from the internet and didn't spot a difference to my other workflows.  [here is an article which describes differences](https://runs-on.com/github-actions/pull-request-vs-pull-request-target/). Well, it wasn't a bug but expected behaviour my bad was that I didn't spot the difference in the trigger name.

**Reply by edemaine** [reply]
Oh, that's exactly what's going on here! Thank you!

**Reply by edemaine** [reply]
Just a followup to confirm that `workflow_dispatch` works around this issue — you can request running within the branch instead of `pull_request_target`'s behavior — so it's a nice way to test `pull_request_target` triggers from within pull requests.  (But you need to modify the main and PR branches to include `workflow_dispatch` triggers.)

**Reply by awingfield916-tech** [reply]
I look forward to checking out the new project

**Comment by SoeAung95** [comment]
Please fix

**Comment by ittybittybit** [comment]
Hi yes Action is in need  can i request for running work flow files please and thank you

**Comment by ngamimatthewsin-droid** [comment]
Thanks for your help I appreciate that and I'll keep it up to date as well I appreciate it.

**Comment by AakashSingh07** [comment]
Thanks for sharing the clarification about the trigger difference between `pull_request` and `pull_request_target`. 

This is a really helpful explanation because it can be easy to miss that small detail when configuring GitHub Actions workflows. The behavior can definitely be confusing when a workflow appears to run an older version of the file.

Your explanation and the reference article make the situation much clearer. Appreciate you pointing out the trigger difference.

**Comment by jagabanamos38-tech** [comment]
091286

**Comment by jagabanamos38-tech** [comment]
![lv_0_20260407020008](https://github.com/user-attachments/assets/a02be414-4156-415d-8fec-6038402011e8)

**Comment by awingfield916-tech** [comment]
Yes

**Comment by 95LightningMcQueen** [comment]


**Reply by awingfield916-tech** [reply]
Look forward to checking it out.
